labs/scorex.md

```diff
@@ -0,0 +1,51 @@
+---
+layout: default
+title: ScoreX
+parent: Labs
+---
+# Lab Name
+ScoreX
+
+# Short Description
+
+ScoreX is a modular blockchain framework built with flexibility in mind to accelerate the design of experimental DTL technologies. ScoreX is the hyperflexible answer to the Fabric and Sawtooth frameworks. ScoreX is developed and maintained by IOHK and the founders of Ergo under the CC0 1.0 license.
+
+# Scope of Lab
+
+Hyperledger products are categorized in four buckets:
+- Distributed ledgers
+- Domain-specific
+- Libraries
+- Tools
+
+ScoreX would fit into the Distributed ledgers as a new framework designed for modularity and experimentation with blockchain technologies. In specific:
+- Sawtooth separates the core system from the application domain.
+- Iroha presents a modular design targeted to IoT projects.
+- Indy focuses on digital identities.
+- Fabric targets general development of applications or solutions and uses a quite rigid design.
+- Burrow focuses on simplicity but still provides the end user a range of interesting choices.
+- Besu supports Ethereum based chains and is trying to make it easy for enterprises to adopt it.
+
+This six-member family is very comprehensive for enterprise needs, but none of them is targeting a very specific (and important) department: Research and Development. ScoreX solves that gap. ScoreX is modular as the frameworks above, but it is built with a high degree of flexibility allowing R&D departments to prototype their own consensus algorithm, transaction design, block design, ledger state management, memory pool structure and ledger rules.
+
+ScoreX is an effective tool for rapid development of new blockchain protocols. While it is still maturing, projects can be launched quickly and effectively. However, this is a tool for experienced developers as it requires real development to reach its full impact. ScoreX can be an effective product for prototyping, proofs-of-concept, or first releases of implementations. The design and architecture is flexible and ScoreX has good potential for future development
+
+Since its inception, ScoreX has tried to explain the framework in a language that appeals to the researchers, as seen in the [ScoreX whitepaper](https://github.com/ScorexFoundation/ScorexTutorial/blob/master/scorex.pdf). The library is built in a functional language, which allows a closer relation between implementation and research. Finally, it is important to note that Blockchain adoption will happen through R&D departments in many enterprises. It is for all these reasons that IOHK believes ScoreX will bring differentiated and important value to Hyperledger.
+
+
+# Initial Committers
+
+- https://github.com/kushti
+- https://github.com/catena2w
+- https://github.com/ceilican
+- https://github.com/oskin1
+- https://github.com/Tolsi
+- https://github.com/daron666
+
+# Sponsor
+
+- Hart Montgomery (hmontgomery@fujitsu.com)
+
+# Pre-existing repository
+
+https://github.com/ScorexFoundation/Scorex
```

Comment:
Hyperledger requires that all contributed code has a DCO sign off. Looking at this repository, it appears that this is not the case. There are two options for fixing this:
1. Amend existing commits in the repository to include DCO sign off
2. Manually transfer the code from the existing repo to the newly created repo within the hyperledger-labs github organization.

Reply:
We can go either way likely, but need to discuss with colleagues

Reply:
@tkuhrt @kushti so as to avoid losing all the git history or altering old commits, I propose we get contributors to explicitly sign [the DCO](https://developercertificate.org/) for their existing commits in some place in the repo. This way we maintain the git history and don't need to alter old commits.

Reply:
Sounds like a plan. @tkuhrt are you ok with this approach?

Reply:
Unfortunately, all commits must have DCO signoff before we can bring over an existing repository in the lab. I have seen others who were able to add the DCO to the commits without modifying the history of the commits.

Reply:
You are able to retroactively add DCO to every commit in history however the commit hashes will change and this may break compatibility with existing downstream users that depend on them. Since this is a project with real downstream users I would like to avoid any disruption. Additionally modifying the people's commits to make it seem as if they were singed off at the time seems more dishonest than them clearly agreeing *now* that their contributions then fell under the DCO.

Reply:
The Linux Foundation (writers of the DCO) also [prefers this approach](https://opensource.stackexchange.com/a/6722) for repos that did not start with signed-off commits.

Reply:
I pushed for the recommended approach in this PR: https://github.com/ScorexFoundation/Scorex/pull/390

Reply:
The DCO signoff on all commits is a requirement of Hyperledger Charter. @ryjones : Can you comment on whether Hyperledger would accept this approach for DCO signoffs for a existing repo that we would transfer to Hyperledger Labs? It was my understanding that are options for DCO signoffs in Hyperledger is one of the following:
1) All commits have DCO signoff and we can bring existing repo over
-or-
2) We create a new repository and the code is brought over with a single commit that has DCO signoff

Reply:
@ryjones you feedback would be appreciated here :)

Reply:
I for one am in favor of reducing barriers to entry. There have to be multiple paths to cure DCO from existing code. We have to think about why DCO is required.
Mostly to handle IP issues, with developers signing off that they have the right to contribute the code that they have checked in and it is free of IP issues. However, this can fail for many reasons including bad faith by the developer who is signing off code that they do not have a right to.
Given the nature of this DCO signoff it is good to have alternate pathways to enforce what is at best an imperfect solution, including a txt file with various people signing off on hashes that they are responsible for checked into the docs folder. I believe the LF has to opine on this, but putting barriers across valuable contributions through strict adherence to the letter and not the spirit of the text is not the way to do this. @gtklocker @tkuhrt  @ryjones @lehors @trbs

Reply:
I agree with Vipin and I think it can be argued that having a text file that lists all the commits by associating their hashes with the relevant signoff still fulfills the requirement. Indeed, "All commits have DCO signoff" doesn't say that the only way this can be done is in the commit log, does it?
Don't get me wrong, I don't think that's ideal and wouldn't want this to continue for future commits but I do think it would fulfill the requirement of ensuring there is a legal commitment for every commit.
For what it's worth, I think this gives us more than a single commit against a new repo which squashes history entirely.

Reply:
Sorry, I missed this. We’ve agreed that you can have a commit that lists the commit ids of your commits and make a DCO statement in that commit. —allow-empty

Reply:
So I believe the ball is in your court @gtklocker @kushti. Or am I wrong?

Reply:
Yes, thank you. @kushti will take it from here. The process of retroactively applying the DCO has started in this PR: https://github.com/ScorexFoundation/Scorex/pull/390

Reply:
@vipinsun @gtklocker @ryjones @lehors @tkuhrt merged!

Reply:
@vipinsun @ryjones @lehors all the issues have been resolved.  Can you please update the review and let us know the next steps.  Thanks!

Reply:
@ahmadalkabra please transfer the existing repo, https://github.com/ScorexFoundation/Scorex, to me.

Reply:
@kushti can you help with this request

Reply:
@ryjones done! "Repository transfer to ryjones requested" (@ahmadalkabra)